core/src/main/scala/org/apache/spark/SparkStatusTracker.scala

```diff
@@ -104,4 +106,34 @@ class SparkStatusTracker private[spark] (sc: SparkContext) {
       }
     }
   }
+
+  /**
+   * Returns a list of all known executors, represented by string with format: "host:port"
+   */
+  def getExecutors(): Array[String] = {
```

Comment:
It seems kind of arbitrary that `getExecutors` returns `host:port` but not IDs. I think it's better that we make a `SparkExecutorInfo` or something and expose the `host:port` there, along with other things like cache size, `numRunningTasks` etc. Then in the future we can add more things we want to expose without tying ourselves with the `host:port` identifier.